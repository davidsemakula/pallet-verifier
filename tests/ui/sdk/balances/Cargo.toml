[package]
name = "pallet-balances"
version = "28.0.0"
#authors.workspace = true
#edition.workspace = true
edition = "2021"
license = "Apache-2.0"
#homepage.workspace = true
#repository.workspace = true
description = "FRAME pallet to manage balances"
readme = "README.md"

#[lints]
#workspace = true

[package.metadata.docs.rs]
targets = ["x86_64-unknown-linux-gnu"]

[dependencies]
codec = { version = "3.6.12", default-features = false, features = ["derive", "max-encoded-len"], package = "parity-scale-codec" }
log = { version = "0.4.22", default-features = false }
scale-info = { version = "2.11.3", default-features = false, features = ["derive"] }
frame-benchmarking = { version = "38.0.0", optional = true, default-features = false }
frame-support = { version = "38.0.0", default-features = false }
frame-system = { version = "38.0.0", default-features = false }
sp-runtime = { version = "39.0.0", default-features = false }
docify = { version = "0.2.8" }

[dev-dependencies]
pallet-transaction-payment = { version = "38.0.0", default-features = true }
frame-support = { version = "38.0.0", default-features = true, features = ["experimental"] }
sp-core = { version = "34.0.0", default-features = true }
sp-io = { version = "38.0.0", default-features = true }
paste = { version = "1.0.15", default-features = true }

[features]
default = ["std"]
std = [
	"codec/std",
	"frame-benchmarking?/std",
	"frame-support/std",
	"frame-system/std",
	"log/std",
	"pallet-transaction-payment/std",
	"scale-info/std",
	"sp-core/std",
	"sp-io/std",
	"sp-runtime/std",
]
# Enable support for setting the existential deposit to zero.
insecure_zero_ed = []
runtime-benchmarks = [
	"frame-benchmarking/runtime-benchmarks",
	"frame-support/runtime-benchmarks",
	"frame-system/runtime-benchmarks",
	"sp-runtime/runtime-benchmarks",
]
try-runtime = [
	"frame-support/try-runtime",
	"frame-system/try-runtime",
	"pallet-transaction-payment/try-runtime",
	"sp-runtime/try-runtime",
]
