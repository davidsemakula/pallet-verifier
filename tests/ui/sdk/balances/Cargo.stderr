warning: possible attempt to add with overflow
  --> src/impl_currency.rs:947:3
   |
LL | /         Self::locks(who)
LL | |             .into_iter()
LL | |             .filter(|l| l.id == id)
LL | |             .fold(Zero::zero(), |acc, l| acc + l.amount)
   | |________________________________________________________^
   |
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/adapters/filter.rs:157:9
   |
LL |         self.iter.fold(init, filter_fold(self.predicate, fold))
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:2640:21
   |
LL |             accum = f(accum, x);
   |                     ^^^^^^^^^^^
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/adapters/filter.rs:41:44
   |
LL |     move |acc, item| if predicate(&item) { fold(acc, item) } else { acc }
   |                                            ^^^^^^^^^^^^^^^
note: related location
  --> src/impl_currency.rs:950:33
   |
LL |             .fold(Zero::zero(), |acc, l| acc + l.amount)
   |                                          ^^^^^^^^^^^^^^
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/arith.rs:110:1
   |
LL | add_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this warning originates in the macro `add_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: possible attempt to subtract with overflow
  --> src/impl_currency.rs:865:6
   |
LL |                     reserves[index].amount -= actual;
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/arith.rs:833:1
   |
LL | sub_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this warning originates in the macro `sub_assign_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: possible attempt to subtract with overflow
  --> src/impl_currency.rs:867:9
   |
LL |                     Ok(value - actual)
   |                        ^^^^^^^^^^^^^^
   |
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2024-02-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/arith.rs:219:1
   |
LL | sub_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this warning originates in the macro `sub_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

